src/commands/run.rs

```diff
@@ -19,18 +21,67 @@ impl Run {
         port: u32,
         filter_search_domain: String,
     ) -> Result<(), Error> {
-        debug!(
-            "Setting up aardvark server with input directory as {:?}",
-            input_dir
-        );
+        // fork and verify if server is running
+        // and exit parent
+        // setsid() ensures that there is no controlling terminal on the child process
 
-        if let Err(er) = serve::serve(&input_dir, port, &filter_search_domain) {
-            return Err(std::io::Error::new(
-                std::io::ErrorKind::Other,
-                format!("Error starting server {}", er),
-            ));
+        match unsafe { fork() } {
+            Ok(ForkResult::Parent { child, .. }) => {
+                log::debug!("starting aardvark on a child with pid {}", child);
+                // verify aardvark here and block till all the ip are ready
+                match config::parse_configs(&input_dir) {
+                    Ok((_, listen_ip_v4, listen_ip_v6)) => {
+                        for (_, ip_list) in listen_ip_v4 {
+                            for ip in ip_list {
+                                serve::wait_till_aardvark_server_ready(
+                                    std::net::IpAddr::V4(ip),
+                                    port,
+                                );
+                            }
+                        }
+                        for (_, ip_list) in listen_ip_v6 {
+                            for ip in ip_list {
+                                serve::wait_till_aardvark_server_ready(
+                                    std::net::IpAddr::V6(ip),
+                                    port,
+                                );
+                            }
+                        }
+                    }
+                    Err(e) => {
+                        return Err(std::io::Error::new(
+                            std::io::ErrorKind::Other,
+                            format!("unable to parse config: {}", e),
+                        ))
+                    }
+                }
+
+                Ok(())
+            }
+            Ok(ForkResult::Child) => {
+                // remove any controlling terminals
```

Comment:
we should close the the stdour/err here and set them to /dev/null instead

Reply:
added in recent push